### ISSUE TITLE
getIndex() proposal: make 'filename' the first arg

### ISSUE THREAD

**Issue by dankelley** [ArgoCanada/argoFloats#94]
TLDR: I plan to switch the order of the first two arguments of `getIndex()`, barring convincing complaints.  (This will not break existing calls that name the `filename` argument, so the docs,  vignette, and videos will be okay.)

The usual rule in R is that the earlier arguments to functions are the ones the user is likely to want to supply, and that the later arguments are the ones for which the defaults are usually appropriate.  Thus, for example, `plot()` has `x` and `y` first, and things like `type`, `xlim`, etc., come later.  This lets users focus on the most important things, and it also reduces the burden of typing out argument names.

When I first wrote `getIndex()`, I was thinking that the `server` would be the thing the user would usually specify, and that `filename` (which was originally called `file`, until I realized that this complicated matters because indices have a column named `file`) would only rarely be specified. 

I was wrong in that assumption.  For one thing, I discovered that servers sometimes do not respond, so I made `getIndex()` sequence through the listed servers.  When I further added the `"auto"` nickname for `server`, it became (I think) unlikely that users would want to specify the `server` argument.  That argues that `server` should not be the first argument. For another thing, it became apparent that the base argo index might not be what people want, some of the time ... they might want "bgc", or they might want "merged" (or, as of yesterday, "synthetic").

Therefore, I now feel strongly that users are more likely to want to specify `filename` than `server`.  That forces typical usage to be e.g. `getIndex(filename="SOMETHING")`, which means that a common use of the function requires a named argument.  That is not typical for R functions, and it can lead to errors if the user thinks that `filename` is the first argument, and thus makes the function call without naming the argument. (I made that error the other day, and even reported an issue ... that's proof that somethings is wrong.)

With this long preamble, I am letting developers and users know that I plan to switch the order of the first two arguments of `getIndex()`.

If I get "OK" comments, or thumbs-up signs from developers, I will make the change over this long weekend.  Otherwise, I'll propose a vf2f at 1000h on Tuesday, to discuss pros and cons, and make a decision.

**Comment by dankelley** [comment]
This has been done, in the "develop" and "master" branches.  Now that it is the first argument there is no longer a need to write e.g. `getIndex(filename="...")` so I changed such lines in the docs and vignette to read e.g `getIndex("...")`.
